lib/url-updater/evaluate-modules.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+const RepoDataClient = require('@financial-times/origami-repo-data-client');
+const semver = require('semver');
+const UserError = require('../utils/usererror');
+
+/**
+ * The result of comparing a requested module / version against all releases.
+ * @typedef {Object} ModuleResult
+ * @property {string} name - The name of the requested module, including npm scope where relevant e.g. "o-table", "@financial-times/o-table".
+ * @property {string} requestedVersion - The version range of the module requested, e.g. "1.0.0", "^1.0.0", "*", ""
+ * @property {string[]} versions - All version numbers for the module, e.g. "2.0.1", "3.0.0-beta.1"
+ * @property {string} latestVersion - The latest version of the module.
+ * @property {boolean} satisfies - Whether the latest version of the module is the version requested, or within its range.
+ * @property {string} latestOrigamiSpec - The version of the Origami Specification which the latest release of the module conforms to.
+ */
+
+/**
+ * @param {Array[]} - An array of arrays representing requested modules, e.g. [['o-example', 'v1.0.0'], 'o-test-component', '']
+ * @return {ModuleResult[]} - Details about the requested modules and how up to date they are.
+ */
+module.exports = async function evaluateModules(modules) {
+    const repoData = new RepoDataClient({});
+    const results = [];
+
+    for await (const module of modules) {
```

Comment:
According to the JSDoc `modules` is an array of arrays, which means we dont need `for await.. of` as we are not iterating over async iterable. We should change this for a `for.. of`.

```suggestion
    for (const module of modules) {
```